glean-core/android/src/main/java/mozilla/telemetry/glean/net/BaseUploader.kt

```diff
@@ -22,6 +22,57 @@ import java.util.TimeZone
 class BaseUploader(d: PingUploader) : PingUploader by d {
     companion object {
         private const val LOG_TAG: String = "glean/BaseUploader"
+        // Since the logcat ring buffer size is configurable, but it's 'max payload' size
+        // is not, we must break apart long pings into chunks no larger than the max payload size
+        // of 4076b.
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        internal const val MAX_LOG_PAYLOAD_SIZE_BYTES = 4000
+
+        /**
+         * Function used to break apart large pings into an array of "chunks" that are compatible with
+         * Logcat's max payload size
+         */
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        internal fun splitPingForLog(pingData: String, path: String): List<String> {
+            // Calculate the total number of chunks
+            var chunkCount = pingData.length / MAX_LOG_PAYLOAD_SIZE_BYTES
```

Comment:
Mh, wait a second: I randomly stumbled upon [`chunked`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/chunked.html), which seems to do exactly this. Maybe we can just use that out of the box?